web_query_timeout/controllers/main.py

```diff
@@ -25,3 +25,15 @@ def do_search_read(self, model, fields=False, offset=0, limit=False,
                 'length': 0,
                 'records': 'WebQueryTimeoutException',
             }
+
+    def _call_kw(self, model, method, args, kwargs):
+        """
+        Set the context value that triggers the database query duration when
+        the triggered method is read_group.
+        """
+        try:
+            res = super(WebQueryTimeoutDataSet, self)._call_kw(
+                model, method, args, kwargs)
+            return res
+        except QueryCanceledError:
+            raise UserError(_('Query cancelled due to timeout'))
```

Comment:
After calling with Pieter, we decided to raise the UserError because otherwise, we had to copy  the group_by method to our own codebase. (this function: https://github.com/odoo/odoo/blob/8.0/addons/web/static/src/js/data.js#L134 )

This way is much cleaner